File: Data Structures & Algorithms/count-number-of-islands/submission-2.cs
public class Solution {
    public int NumIslands(char[][] grid) {
        int result = 0;

        for(int i =0;i<grid.Length;i++){
            for(int j =0;j<grid[0].Length;j++){
                if(grid[i][j] == '1'){
                    DFS(grid, i, j);
                    result++;
                }
            }
        }
        return result;
        
    }

    private void DFS(char[][] grid, int i, int j){
        if(i < 0 || j < 0 || i >= grid.Length || j >= grid[0].Length || grid[i][j] != '1'){
            return;
        }
        grid[i][j] = '0';

        DFS(grid, i, j - 1);
        DFS(grid, i, j + 1);
        DFS(grid, i - 1, j);
        DFS(grid, i + 1, j);
    }
}

File: Data Structures & Algorithms/search-for-word/submission-2.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:
        for i in range(len(board)):
            for j in range(len(board[0])):
                val = self.dfs(board, word, i, j, 0)
                if val:
                    return True
        return False
    
    def dfs(self, board, word, i, j, idx):
        if idx >= len(word):
            return True
        
        if i < 0 or i >= len(board) or j < 0 or j>= len(board[0]) or board[i][j] != word[idx]:
            return False
        return self.dfs(board, word, i+1,j,idx+1) or self.dfs(board, word, i, j+1, idx + 1)